src/Client/ClientBase.cpp

```diff
@@ -549,11 +549,26 @@ try
                             range.second);
                 }
 
-                out_file_buf = wrapWriteBufferWithCompressionMethod(
-                    std::make_unique<WriteBufferFromFile>(out_file, DBMS_DEFAULT_BUFFER_SIZE, O_WRONLY | O_EXCL | O_CREAT),
-                    compression_method,
-                    compression_level
-                );
+                if (query_with_output->is_into_outfile_with_stdout)
```

Comment:
Ok, but maybe we can avoid duplication?
Create the main write buffer, and if needed - move it to the ForkWriteBuffer?

Reply:
i have addressed this. thanks